app/code/local/VF/CustomMenu/Block/Navigation.php

```diff
@@ -50,6 +51,70 @@ protected function _construct()
                 VF_CustomMenu_Model_Menu::CACHE_TAG
             ),
         ));
+
+        $this->getCategoryUrls();
+    }
+
+    /**
+     * Generate the dynamic category URL's and cache them.
```

Comment:
some more info would be really helpful (What is dynamic category and why it is needed).
I have to read through the whole function to understand what is does.